osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -311,6 +313,24 @@ Task IMultiplayerClient.UserStateChanged(int userId, MultiplayerUserState state)
             return Task.CompletedTask;
         }
 
+        Task IMultiplayerClient.UserBeatmapAvailabilityChanged(int userId, BeatmapAvailability beatmapAvailability)
+        {
+            if (Room == null)
+                return Task.CompletedTask;
+
+            Scheduler.Add(() =>
+            {
+                if (Room == null)
+                    return;
+
+                Room.Users.Single(u => u.UserID == userId).BeatmapAvailability = beatmapAvailability;
```

Comment:
should be tolerant against user potentially not being in the room. we don't really ever care if this fails.

Reply:
I'm not sure about that, `UserLeft` will remove the user immediately (technically in the main thread but whatever), there shouldn't be any way at all that availability is changed in a user that is not in the room anymore, and sounds like a nasty bug that needs fixing.

Reply:
this is for display only. if the user has somehow left the room already it's way worse to have this throw an error than not.

Reply:
hmm, you have a point, I'm fine with your suggestion, but at the very least logging the error such that it goes to sentry would be a better idea, because this sounds like it should never be the case. sounds good?

Reply:
if you're going to go that far, you should be arguing that the `room == null` case should also be reported. i don't think either need to be. reporting to sentry should be cases where the game is in a state it cannot continue (ie. something we need to fix). logging things like this just adds noise.

Reply:
oh, my bad. I see.